code/datums/components/storage/storage.dm

```diff
@@ -523,6 +523,8 @@
 /datum/component/storage/proc/can_be_inserted(obj/item/I, stop_messages = FALSE, mob/M)
 	if(!istype(I) || (I.flags_1 & ABSTRACT_1))
 		return FALSE //Not an item
+	if(I == src.parent)

```

Comment:
***s r c***

Reply:
it's needed to distinguish from the var/atom/parent = src.parent below. otherwise i believe it will compile error @ninjanomnom

Reply:
That's not a problem but the better choice there is don't have a var locally defined that uses the same name as one on the type.